python/GafferSceneUI/SceneViewUI.py

```diff
@@ -669,6 +641,66 @@ def __menuDefinition( self ) :
 		m.append( "/Expand All Divider", { "divider" : True } )
 		m.append( "/Expand All", { "checkBox" : expandAll, "command" : Gaffer.WeakMethod( self.__toggleMinimumExpansionDepth ) } )
 
+		m.append( "/PurposesDivider", { "divider" : True, "label" : "Purpose" } )
+
+		# \todo Move the `includedPurposes` plug out of `drawingMode` and put it on a new plug that holds (and replaces)
+		# `minimumExpansionDepth`.
+		drawingModePlug = self.getPlug().parent()["drawingMode"]
+		includedPurposes = drawingModePlug["includedPurposes"]["value"].getValue()
+		includedPurposesEnabled = drawingModePlug["includedPurposes"]["enabled"].getValue()
+
+		renderElements = [ "default", "render" ]
+		previewElements = [ "default", "proxy" ]
+		previewWithGuidesElements = [ "default", "proxy", "guide" ]
+		m.append(
+			"/Render",
+			{
+				"checkBox" : set( includedPurposes ) == set( renderElements ),
+				"active" : includedPurposesEnabled,
+				"command" : functools.partial( drawingModePlug["includedPurposes"]["value"].setValue, IECore.StringVectorData( renderElements ) )
+			}
+		)
+		m.append(
+			"/Preview",
+			{
+				"checkBox" : set( includedPurposes ) == set( previewElements ),
+				"active" : includedPurposesEnabled,
+				"command" : functools.partial( drawingModePlug["includedPurposes"]["value"].setValue, IECore.StringVectorData( previewElements ) )
+			}
+		)
+		m.append(
+			"/Preview with Guides",
+			{
+				"checkBox" : set( includedPurposes ) == set( previewWithGuidesElements ),
+				"active" : includedPurposesEnabled,
+				"command" : functools.partial( drawingModePlug["includedPurposes"]["value"].setValue, IECore.StringVectorData( previewWithGuidesElements ) )
+			}
+		)
+
+		allPurposes = [ "default", "render", "proxy", "guide" ]
+		for purpose in allPurposes :
+			newPurposes = IECore.StringVectorData( [
+				p for p in allPurposes
+				if
+				( p != purpose and p in includedPurposes ) or ( p == purpose and p not in includedPurposes )
+			] )
+			m.append(
+				"/Values/{}".format( purpose.capitalize() ),
+				{
+					"checkBox" : purpose in includedPurposes,
+					"active" : includedPurposesEnabled,
+					"command" : functools.partial( drawingModePlug["includedPurposes"]["value"].setValue, newPurposes ),
+				}
+			)
+			m.append( "/Values/SceneDivider", { "divider" : True } )
+			m.append(
+				"/Values/From Scene",
+				{
+					"checkBox" : not includedPurposesEnabled,
+					"command" : lambda checked : drawingModePlug["includedPurposes"]["enabled"].setValue( not checked ),
+				}
+			)
+
```

Comment:
The "Expand USD Instancers" menu item is ending up under this one, but it feels like it would be better off just under "Expand All" (perhaps even without the divider?). Hopefully this should be doable using `insertAfter()` in `startup/GafferSceneUI/usdPointInstancerAdaptor.py`.